src/Servers/Kestrel/Core/src/HttpsConnectionAdapterOptions.cs

```diff
@@ -39,6 +39,11 @@ public HttpsConnectionAdapterOptions()
         /// </summary>
         public X509Certificate2 ServerCertificate { get; set; }
 
+        /// <summary>
+        /// Specifies the intermediate certificates in the chain.
+        /// </summary>
+        public X509Certificate2Collection ServerCertificateIntermediates { get; set; }
```

Comment:
The weird thing about this property is that it should *really* be set with the `ServerCertificate` in an atomic manner. If the  cert is set, then the intermediates should be set as well.

Reply:
Agreed. The combined type should probably be defined in the runtime.

Reply:
Should we add a `SslStreamCertificateContext` property instead of this `X509Certificate2Collection`?